src/ert/shared/models/ensemble_smoother.py

```diff
@@ -24,6 +34,175 @@ def setAnalysisModule(self, module_name: str) -> None:
         if not module_load_success:
             raise ErtRunError(f"Unable to load analysis module '{module_name}'!")
 
+    async def run(self, evaluator_server_config: EvaluatorServerConfig) -> None:
+        loop = asyncio.get_running_loop()
+        threadpool = concurrent.futures.ThreadPoolExecutor()
+
+        prior_context = await loop.run_in_executor(threadpool, self.create_context)
+
+        experiment_logger.debug("starting ensemble smoother experiment")
+        await self.dispatch(
+            CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_STARTED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        self._checkMinimumActiveRealizations(prior_context)
```

Comment:
`loop.run_in_executor(` ? Not sure where should be a threshold for saying this function is "atomic" does not need to be async.

Reply:
Tracing through the code, this seems to only involve trivial getters and an integer comparison. My gut feeling is then not to  
yield through await since there might be some overhead with our switching between sync and async context? (this is partly speculation)

Reply:
You are right there actually! For reference: https://stackoverflow.com/questions/54813458/asyncio-performance-with-synchronous-code